rust/src/http2/http2.rs

```diff
@@ -0,0 +1,892 @@
+                                        tx.type_data = Some(HTTP2FrameTypeData::WINDOWUPDATE(wu));
+                                    }
+                                    Err(nom::Err::Incomplete(_)) => {
+                                        return AppLayerResult::incomplete(
+                                            (il - input.len()) as u32,
+                                            (HTTP2_FRAME_HEADER_LEN + HTTP2_FRAME_WINDOWUPDATE_LEN)
+                                                as u32,
+                                        );
+                                    }
+                                    Err(_) => {
+                                        self.set_event(HTTP2Event::InvalidFrameData);
+                                    }
+                                }
+                            }
+                        }
+                        parser::HTTP2FrameType::PUSHPROMISE => {
+                            //we need to check for completeness first
+                            if rem.len() < hl {
+                                return AppLayerResult::incomplete(
+                                    (il - input.len()) as u32,
+                                    (HTTP2_FRAME_HEADER_LEN + hl) as u32,
+                                );
+                            }
+                            match parser::http2_parse_frame_push_promise(
+                                &rem[..hl],
+                                head.flags,
+                                &mut self.dynamic_headers,
+                            ) {
+                                Ok((_, hs)) => {
+                                    for i in 0..hs.blocks.len() {
+                                        if hs.blocks[i].error >= parser::HTTP2HeaderDecodeStatus::HTTP2HeaderDecodeError {
+                                            self.set_event(HTTP2Event::InvalidHeader);
+                                        }
+                                    }
+                                    //TODO6tx right transaction wih promised headers
+                                    tx.type_data = Some(HTTP2FrameTypeData::PUSHPROMISE(hs));
+                                }
+                                Err(_) => {
+                                    self.set_event(HTTP2Event::InvalidFrameData);
+                                }
+                            }
+                        }
+                        parser::HTTP2FrameType::DATA => {
+                            //we check for completeness first for code simplicity
+                            if rem.len() < hl {
+                                return AppLayerResult::incomplete(
+                                    (il - input.len()) as u32,
+                                    (HTTP2_FRAME_HEADER_LEN + hl) as u32,
+                                );
+                            }
+                            //TODOask use streaming buffer directly
```

Comment:
Is everything a file ?
Or data sent to the server should go to a streaming buffer directly ? (as in HTTP 1)